playground/backend/internal/code_processing/code_processing.go

```diff
@@ -189,8 +189,13 @@ func Process(ctx context.Context, cacheService cache.Cache, lc *fs_tool.LifeCycl
 		return
 	}
 	if !ok {
-		if runOutput.Error != nil {
-			runError.Write([]byte(runOutput.Error.Error()))
+		// If unit test has some error then error output is placed as RunOutput
+		if isUnitTest {
+			output, err := GetProcessingOutput(ctx, cacheService, pipelineId, cache.RunOutput, "")
+			if err == nil {
+				runError.Write([]byte(output))
+			}
```

Comment:
Don't we need to process this `err` if it is not `nil`?

Reply:
This error occurs only if `GetProcessingOutput` method couldn't receive the correct run output (there is no run output in the cache or it is no string value). In this case, `GetProcessingOutput` method logs this error.

Receiving this error we do not need to send it to the other methods or to log it, so we just will not use this error, because on the next lines we call `processRunError` method after which the whole code processing will be finished.